config/configcompression/compressiontype.go

```diff
@@ -25,19 +37,45 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == TypeLz4 ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (ct *TypeWithLevel) UnmarshalText(in []byte) error {
+	var err error
+	parts := strings.Split(string(in), "/")
+	compressionTyp := Type(parts[0])
+	level := zlib.DefaultCompression
+	if len(parts) == 2 {
+		level, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return fmt.Errorf("invalid compression level: %q", parts[1])
+		}
+		if compressionTyp == TypeSnappy ||
+			compressionTyp == TypeLz4 ||
+			compressionTyp == typeNone ||
+			compressionTyp == typeEmpty {
+			return fmt.Errorf("compression level is not supported for %q", compressionTyp)
+		}
+	}
+	ct.Level = Level(level)
+	if (compressionTyp == TypeGzip && isValidLevel(level)) ||
+		(compressionTyp == TypeZlib && isValidLevel(level)) ||
+		(compressionTyp == TypeDeflate && isValidLevel(level)) ||
+		compressionTyp == TypeSnappy ||
+		compressionTyp == TypeLz4 ||
+		compressionTyp == TypeZstd ||
+		compressionTyp == typeNone ||
+		compressionTyp == typeEmpty {
+		ct.Level = Level(level)
+		ct.Type = compressionTyp
 		return nil
 	}
-	return fmt.Errorf("unsupported compression type %q", typ)
 
+	return fmt.Errorf("unsupported compression type and level(default if not specified) %s - %d", compressionTyp, ct.Level)
+
+}
+
+func isValidLevel(level int) bool {
+	return level == zlib.BestSpeed ||
+		level == zlib.BestCompression ||
+		level == zlib.DefaultCompression ||
+		level == zlib.HuffmanOnly ||
+		level == zlib.NoCompression
```

Comment:
```suggestion
	// https://pkg.go.dev/compress/gzip#NewWriterLevel says:
	//     "The compression level can be DefaultCompression, NoCompression, HuffmanOnly
	//     or any integer value between BestSpeed and BestCompression inclusive."
	return level == zlib.DefaultCompression ||
			level == zlib.NoCompression ||
			level == zlib.HuffmanOnly ||
			(level >= zlib.BestSpeed && level <= zlib.BestCompression)
```

Reply:
The compression levels seem to be discrete values, which is why I didn't use a range.

Reply:
@rnishtala-sumo please see the comment I included above, which is taken directly from the compress/gzip package docs. `BestSpeed-BestCompression (1-9)` is a range, while `DefaultCompression` and `NoCompression` have special meaning.

Reply:
the collector uses this library where they are discrete - https://pkg.go.dev/github.com/klauspost/compress/gzip#pkg-constants

Reply:
The same comment exists on https://pkg.go.dev/github.com/klauspost/compress/gzip#NewWriterLevel. It's a standard part of zlib.

Reply:
I made the change.